bundles/CoreBundle/Resources/config/class_builder.yaml

```diff
@@ -0,0 +1,63 @@
+services:
+    _defaults:
+        public: true
+
+    Pimcore\DataObject\ClassBuilder\ClassBuilderInterface: '@Pimcore\DataObject\ClassBuilder\ClassBuilder'
+    Pimcore\DataObject\ClassBuilder\ClassBuilder:
+        arguments:
+            - '@Pimcore\DataObject\ClassBuilder\FieldDefinitionDocBlockBuilderInterface'
+            - '@Pimcore\DataObject\ClassBuilder\FieldDefinitionPropertiesBuilderInterface'
+            - '@Pimcore\DataObject\ClassBuilder\FieldDefinitionBuilderInterface'
+
+    Pimcore\DataObject\ClassBuilder\FieldCollectionClassBuilderInterface: '@Pimcore\DataObject\ClassBuilder\FieldCollectionClassBuilder'
+    Pimcore\DataObject\ClassBuilder\FieldCollectionClassBuilder:
+        arguments:
+            - '@Pimcore\DataObject\ClassBuilder\FieldDefinitionDocBlockBuilderInterface'
+
+    Pimcore\DataObject\ClassBuilder\FieldDefinitionBuilderInterface: '@Pimcore\DataObject\ClassBuilder\FieldDefinitionBuilder'
+    Pimcore\DataObject\ClassBuilder\FieldDefinitionBuilder: ~
+
+    Pimcore\DataObject\ClassBuilder\FieldDefinitionDocBlockBuilderInterface: '@Pimcore\DataObject\ClassBuilder\FieldDefinitionDocBlockBuilder'
+    Pimcore\DataObject\ClassBuilder\FieldDefinitionDocBlockBuilder: ~
+
+    Pimcore\DataObject\ClassBuilder\FieldDefinitionPropertiesBuilderInterface: '@Pimcore\DataObject\ClassBuilder\FieldDefinitionPropertiesBuilder'
+    Pimcore\DataObject\ClassBuilder\FieldDefinitionPropertiesBuilder: ~
+
+    Pimcore\DataObject\ClassBuilder\ListingClassBuilderInterface: '@Pimcore\DataObject\ClassBuilder\ListingClassBuilder'
+    Pimcore\DataObject\ClassBuilder\ListingClassBuilder:
+        arguments:
+            - '@Pimcore\DataObject\ClassBuilder\ListingClassFieldDefinitionBuilderInterface'
+
+    Pimcore\DataObject\ClassBuilder\ListingClassFieldDefinitionBuilderInterface: '@Pimcore\DataObject\ClassBuilder\ListingClassFieldDefinitionBuilder'
+    Pimcore\DataObject\ClassBuilder\ListingClassFieldDefinitionBuilder: ~
+
+    Pimcore\DataObject\ClassBuilder\ObjectBrickClassBuilderInterface: '@Pimcore\DataObject\ClassBuilder\ObjectBrickClassBuilder'
+    Pimcore\DataObject\ClassBuilder\ObjectBrickClassBuilder:
+        arguments:
+            - '@Pimcore\DataObject\ClassBuilder\FieldDefinitionDocBlockBuilderInterface'
+
+    Pimcore\DataObject\ClassBuilder\ObjectBrickContainerClassBuilderInterface: '@Pimcore\DataObject\ClassBuilder\ObjectBrickContainerClassBuilder'
+    Pimcore\DataObject\ClassBuilder\ObjectBrickContainerClassBuilder: ~
```

Comment:
I'd prefer to not have service aliases, as they are also public then by default, anything that's against using just the "interface"-services? 
So: 
```yaml
Pimcore\DataObject\ClassBuilder\ObjectBrickContainerClassBuilderInterface:
    class: Pimcore\DataObject\ClassBuilder\ObjectBrickContainerClassBuilder
```

Reply:
this is basically the way I always do it in Symfony. I think it is also the Symfony way of doing it... Can change though if you want.

Reply:
Well, this case is probably special, because here we have both `ObjectBrickContainerClassBuilderInterface` and `ObjectBrickContainerClassBuilder` `public` ... and the latter shouldn't be public imho.

Reply:
Yeah, would make sense, as it is best practice and afaik `public` services are also more expensive.

Reply:
So only this is left then, the rest looks good to me. Thanks!

Reply:
changed

Reply:
changed

Reply:
changed